#! /usr/bin/env python

PACKAGE='nuturtle_robot'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name                qType      Level
#       Description
#       Default     Min   Max
gen.add("k_rot_comp",          double_t,      0,
        "to compensate for the additional friction in rotation",
		        0.0, 1, 2.0)
gen.add("max_trans_vel",          double_t,      0,
        "maximum translational velocity",
        0.5, 0.0, 1.0)
gen.add("max_wheel_command",          double_t,      0,
        "maximum pwm",
        32.0, 0.0, 32.0)
gen.add("cmd_ang_vel",          double_t,      0,
        "cheat angular velocity",
        0.7, 0.0, 1.0)
gen.add("cmd_trans_vel",          double_t,      0,
        "cheat translational velocity",
        0.2, 0.0, 0.5)
gen.add("ang_k",          double_t,      0,
        "cheat angular velocity k",
        3.0, 0.0, 5.0)
# gen.add("min_pwm",          int_t,      0,
#         "minimum pwm for the robot",
#         14, 32, 32)

exit(gen.generate(PACKAGE, "turtle_interface", "TurtleInterface"))

